File: src/main/java/com/epicsquid/thermalendergy/init/ThermalEndergyLang.java
package com.epicsquid.thermalendergy.init;

import com.epicsquid.thermalendergy.ThermalEndergy;
import com.tterrag.registrate.Registrate;
import net.minecraft.network.chat.Component;

public class ThermalEndergyLang {


	private static final Registrate REGISTRATE = ThermalEndergy.registrate();

	public static final Component CREATIVE_TAB = REGISTRATE.addRawLang("itemGroup." + ThermalEndergy.MODID, "Thermal Endergy");

	public static void classload() {
	}
}

File: src/main/kotlin/dev/epicsquid/thermalendergy/data/ThermalEndergyLang.kt
package dev.epicsquid.thermalendergy.data

import dev.epicsquid.thermalendergy.ThermalEndergy
import dev.epicsquid.thermalendergy.registry.BlockRegistry
import net.minecraft.data.PackOutput
import net.minecraftforge.common.data.LanguageProvider

class ThermalEndergyLang(
	output: PackOutput,
	locale: String
) : LanguageProvider(output, ThermalEndergy.MODID, locale) {

	override fun addTranslations() {
		add(BlockRegistry.melodiumBlock, "Melodium Block")
		add(BlockRegistry.prismaliumBlock, "Prismalium Block")
		add(BlockRegistry.stellariumBlock, "Stellarium Block")

		add("itemGroup.${ThermalEndergy.MODID}", "Thermal Endergy")
	}
}